libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -225,6 +226,31 @@ static void magnetic_cb1(const uavcan::ReceivedDataStructure<uavcan::equipment::
 {   magnetic_cb(msg, 1); }
 static void (*magnetic_cb_arr[2])(const uavcan::ReceivedDataStructure<uavcan::equipment::ahrs::MagneticFieldStrength>& msg)
         = { magnetic_cb0, magnetic_cb1 };
+        
+static void magnetic_cb_2(const uavcan::ReceivedDataStructure<uavcan::equipment::ahrs::MagneticFieldStrength2>& msg, uint8_t mgr)
+{
+    if (hal.can_mgr[mgr] != nullptr) {
+        AP_UAVCAN *ap_uavcan = hal.can_mgr[mgr]->get_UAVCAN();
+        if (ap_uavcan != nullptr) {
+            AP_UAVCAN::Mag_Info *state = ap_uavcan->find_mag_node(msg.getSrcNodeID().get(), msg.sensor_id);
+            if (state != nullptr) {
+                state->mag_vector[0] = msg.magnetic_field_ga[0];
+                state->mag_vector[1] = msg.magnetic_field_ga[1];
+                state->mag_vector[2] = msg.magnetic_field_ga[2];
+
+                // after all is filled, update all listeners with new data
+                ap_uavcan->update_mag_state(msg.getSrcNodeID().get(), msg.sensor_id);
+            }
+        }
+    }
+}
+
+static void magnetic_cb_2_0(const uavcan::ReceivedDataStructure<uavcan::equipment::ahrs::MagneticFieldStrength2>& msg)
+{   magnetic_cb_2(msg, 0); }
```

Comment:
this whitespace on these functions is strange and makes it hard to read, can you fix it please?

Reply:
ahh, I see. You copied the above code example which is equally weird looking. Fair enough. We can clean it up later on another PR.